modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/GroupUpdateRequest.java

```diff
@@ -254,42 +292,60 @@ private static CompletableFuture<Integer> partitionUpdate(
             return completedFuture(ASSIGNMENT_NOT_UPDATED.ordinal());
         }
 
-        Iif invokeClosure;
-
-        if (aliveStableNodes.isEmpty()) {
-            if (!manualUpdate) {
-                return completedFuture(ASSIGNMENT_NOT_UPDATED.ordinal());
-            }
+        if (aliveStableNodes.isEmpty() && !manualUpdate) {
+            return completedFuture(ASSIGNMENT_NOT_UPDATED.ordinal());
+        }
 
+        if (manualUpdate) {
             enrichAssignments(partId, aliveDataNodes, replicas, partAssignments);
+        }
 
-            // There are no known nodes with data, which means that we can just put new assignments into pending assignments with "forced"
-            // flag.
-            invokeClosure = prepareMsInvokeClosure(
-                    partId,
-                    longToBytesKeepingOrder(revision),
-                    Assignments.forced(partAssignments, assignmentsTimestamp).toBytes(),
-                    null
-            );
-        } else {
-            Set<Assignment> stableAssignments = Set.copyOf(partAssignments);
+        Assignment nextAssignment = aliveStableNodes.isEmpty()
+                ? nextAssignment(partAssignments)
```

Comment:
It worth using `nextAssignment(localPartitionStateMessageByNode, partAssignments)` even in case of `aliveStableNodes.isEmpty()` because you may enrich nodes with pending ones that also may have some data.